### PR TITLE
Chore: kvm list page component refactor MAASENG-1816

### PR DESCRIPTION
**Pull request by Jay-Topher**
## Done

- Refactor KVMList to use PageContent

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

## Fixes
[MAASENG-1816](https://warthogs.atlassian.net/browse/MAASENG-1816)

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots

It could be helpful to provide some screenshots to aid in QAing the change.

[MAASENG-1816]: https://warthogs.atlassian.net/browse/MAASENG-1816?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/kvm/components/KVMHeaderForms/index.ts
Comment (ndv99) [inline src/app/kvm/components/KVMHeaderForms/index.ts:1]: Don't forget to rename the folder here as well!

**Review by ndv99 — COMMENTED** [review]
Code looks good on this, just make sure to fix the failing test cases as well

**Comment by ndv99** [comment]
```
SassError: Can't find stylesheet to import.
    ╷
183 │ @import "~app/kvm/components/KVMHeaderForms/ComposeForm/InterfacesTable";
    │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ╵
  src/scss/index.scss 183:9  root stylesheet
```

Remember to update the stylesheet too :wink:

**Comment by tmerten** [comment]
Seems like there is still a problem with typing on build.

**Comment by petermakowski** [comment]
Could you please resolve the remaining issues including merge conflicts @Jay-Topher

File: src/app/base/sagas/websockets.test.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { eventChannel } from "redux-saga";
+import { call, put, take } from "redux-saga/effects";
```
Comment (petermakowski) [inline src/app/base/sagas/websockets.test.ts:2]: There are multiple lint errors in this PR. Could you please fix.

**Review by petermakowski — COMMENTED** [review]
Could you please merge back latest changes from sidepanel-refactor branch to resolve conflicts.

**Review by petermakowski — COMMENTED** [review]
There are some merge conflicts that need solving please. @Jay-Topher

File: src/app/Routes.tsx [hunk 2]
```diff
@@ -191,10 +199,18 @@ const Routes = (): JSX.Element => (
...
+        path={`${urls.images.index}/*`}
+      />
+      <Route
+        element={
+          <ErrorBoundary>
+            <MachineDetails />
           </ErrorBoundary>
         }
-        path={`${urls.kvm.index}/*`}
+        path={`${urls.machines.machine.index(null)}/*`}
```
Comment (petermakowski) [inline src/app/Routes.tsx:213]: Looks like a mistake from resolving merge conflicts, could you please remove.

These have been moved to use the new page wrapper, see the lines above: https://github.com/canonical/maas-ui/pull/4989/files#diff-cf93f5edcf06a2c153726ff6ffd1514499db8618fe0592bbee81f0fa5a8887faR151

**Pull request merged**